Application port tutorial5

danpham02 opened the pull request:
most modules are coded the same in tutorial 5, they are progressed when the app loses and regains focus.

ssah0022 committed:
Created page for tutorial 5

zgom0003 committed:
Added link to open_voice_command.dart

zgom0003 committed:
Initial commit of open_voice_command.dart (template from tute 2)

zgom0003 committed:
Added intro to open_voice_command module, adding instructions after callback
next.

zgom0003 committed:
Was getting an error about how the build.gradle was allwoing
use of older andriod sdk versions. Made recommended changes
and the application runs fine now.

lmaa0022 committed:
Added Modules for go home and open recent

zgom0003 committed:
Merge branch 'ApplicationPortTutorial5' of https://github.com/Monash-FIT3170/Team4-Android-Talkback into ApplicationPortTutorial5

zgom0003 committed:
-Added basic functionality for open_voice_command.dart
from tutorial 5.
-pubspec.lock has the flutter_tts dependency added to it

Kanishkmon0006 committed:
-Adding open_notification.dart. Module for tute 5
-Adding link to open_notification module in tute 5

Kanishkmon0006 committed:
Merge branch 'ApplicationPortTutorial5' of https://github.com/Monash-FIT3170/Team4-Android-Talkback into ApplicationPortTutorial5

Kanishkmon0006 committed:
-added implementation for open_notification.dart for tut 5
-removed extra button in tut 5

Kanishkmon0006 committed:
-added implementation for open_talkbackmenu for tut 5
-added link to open_talkbackmenu in tut 5

lmaa0022 committed:
open_recent_prototype

lmaa0022 committed:
Editted and fixed GoHome and OpenRecent Modules

zgom0003 committed:
-Refactored code to exclued the calls to tts in the scaffold.
The tts is called in the widget before the scaffold

zgom0003 committed:
Merge branch 'ApplicationPortTutorial5' of https://github.com/Monash-FIT3170/Team4-Android-Talkback into ApplicationPortTutorial5

zgom0003 committed:
-Compeleted open_voice_command module.

zgom0003 committed:
- UI for both go_home and open_recent modules were
updated to be more similar to the original application
- There was a bug where the app skiped the second stage
and directly entered the thire, fxed by reordering 'if'
statements in changeStage() method

zgom0003 committed:
- The pub files now include flutter_tts

danpham02 committed:
rough port of open notifcation and open talkback menu modules

lmaa0022 committed:
Added text display for go_home module

lmaa0022 committed:
Added text display for open_notifications module

lmaa0022 committed:
Added open_recent module's text display

lmaa0022 committed:
Added text display for open_talkback_menu module

byebrid commented:
Need to go now, but here are some initial notes

## Open recent apps
- Unclear what exactly user should do here. I swiped left and up and indeed saw list of recent apps. But returning to the app doesn't do anything. Is it meant to automatically finish the module?
- Small typo

## Go to home screen
- This one worked better!
- Some repetitive wording in first part.
- Text formatting a bit odd, but perhaps we can worry about this later when we start using files to store text. Issue is newlines spaces are preserved in the text element.
- Don't get sent to main menu upon finishing?

byebrid commented:
## Open notification
- Not clear how to *leave* notifications once you open it, based on instructions. Should add to instruction mentioning that you can swipe up from the bottom.
- No way to complete the module? Can open up notification panel, but then nothing else happens.

## Open talkback menu
- No way to complete module.

## Voice command
- Screen is empty, and I don't hear anything. I noticed this error in my log:
```
D/TTS     (22069): Utterance ID has been stopped: 5f7eb146-c078-47c8-953a-88017ba613d3. Interrupted: true
```
- I managed to hear the starting instructions once. I believe it's when flutter hot-reloaded while I was already in the module. I suspect that some talkback is getting triggered as you open the module which then interrupts the text-to-speech.
- To fix this, I think just use text elements like you have with the other modules. Unless there was a specific reason to stick with the text-to-speech package.

## Other notes
- Style a bit inconsistent with other modules, but let's leave this for team discussion.
- For larger instruction prompts, it might be worth breaking it up into multiple text elements? That way a user can more quickly re-listen to a particular section of the prompt, rather than having to sit through the whole prompt just to hear the part they want to hear.
- Will need to double-check merge conflicts. Be careful with the dependency specs `pubspec.*`.
- I tested on Pixel 4 API 34 android-x86 emulator.

Commit:
initial commit for tutorial 5 challenge

danpham02 committed:
remade open_notification module

danpham02 committed:
remade open_recent module

danpham02 committed:
remade go_home and open_talkback_menu

danpham02 committed:
remade open_voice_command module, not finished

danpham02 commented:
I've remade every module in tutorial 5 besides the challenge and the voice_command, module (which can just be redesigned to be more simple) should detect window change on using notification tray now. Tts has been removed and modules use tutorial_main as a base. Make sure you are using the latest flutter version because one of the classes used is a relatively new feature.

Commit:
Added notification handeling class for tut5

Commit:
Added end of lesson challenge

Commit:
Merge branch 'ApplicationPortTutorial5' of https://github.com/Monash-FIT3170/Team4-Android-Talkback into ApplicationPortTutorial5

Commit:
commit before upgrading flutter

Commit:
Added end on lesson challenge one. This challenge combines the notification and scrolling skills.

Commit:
Completed part one of challenge, There is an issue where clicking on the push notification redirects user but clicking on the notification in the drawer does not

Kanishkmon0006 committed:
Added second half of tutorial 5 challenge

Kanishkmon0006 committed:
Merge branch 'ApplicationPortTutorial5' of https://github.com/Monash-FIT3170/Team4-Android-Talkback into ApplicationPortTutorial5

Commit:
Added end of challenge to route

Commit:
adding pubspec

Commit:
Merge branch 'ApplicationPortTutorial5' of https://github.com/Monash-FIT3170/Team4-Android-Talkback into ApplicationPortTutorial5

Commit:
Removed unused import

Commit:
Initial commit for knots&crosses game skeleton

Kanishkmon0006 committed:
Added game intitialization,make move and is board
full

Commit:
Deleted old modules

Commit:
Merge branch 'ApplicationPortTutorial5' of https://github.com/Monash-FIT3170/Team4-Android-Talkback into ApplicationPortTutorial5

lmaa0022 committed:
Added method checkForWin and resetGame, also implemented Widget buildCell

Commit:
Merge branch 'ApplicationPortTutorial5' of https://github.com/Monash-FIT3170/Team4-Android-Talkback into ApplicationPortTutorial5

Commit:
Added basic talkback functionality. The talkback reads out the row and column index

Commit:
Added opponent movements. Added tts for opponent movements. Added tts for win/draw announcement

byebrid committed:
Minor fixes to Tutorial 5 challenge

- Remove Billie's message from instruction text in Tut 5 end of lesson challenge. This was only meant to be in the notification itself.
- Prevent instruction text from being displayed in tut 5 challenge part 2.
- Fix display of tut 5 end of lesson challenge buttons page

byebrid committed:
Merge branch 'main' into ApplicationPortTutorial5

ssah0022 committed:
Merge branch 'main' into ApplicationPortTutorial5
